### PR TITLE
Setup linter

### LINKED ISSUE #5: Enable markdown linting
To ensure correctness and consistency across markdown files.

### PR DESCRIPTION
**Pull request by christoferlof**
Closes #5 
Action created and configured to run on pushes and pull requests. 
Only markdown enabled for now, enable additional linters via config as we need them.

### CODE REVIEW

File: .github/workflows/linter.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+on:
+  push:
+    branches-ignore: [main]
+  pull_request:
+    branches: [main]
+
+  workflow_dispatch:
+
+jobs:
+  build:
```
Comment (marrobi) [inline .github/workflows/linter.yml:12]: Should this be lint as we aren't actually building anything? Just makes the output make more sense.

**Review by marrobi — COMMENTED** [review]
Looking at the run, it's not actually run on any files:

image

Any chance can get a pass and a fail on an actual file to verify its working as expected?

File: .github/workflows/linter.yml (hunk 1 above)
Comment (marrobi) [reply .github/workflows/linter.yml:12]: Ran using act locally on all files, should these errors be fixed prior to merge:

```
| 2021-03-05 14:34:49 [INFO]   File:[/github/workspace/README.md]
| 2021-03-05 14:34:49 [ERROR]   Found errors in [markdownlint] linter!
| 2021-03-05 14:34:49 [ERROR]   Error code: 1. Command output:
| ------
| /github/workspace/README.md:16:57 MD034/no-bare-urls Bare URL used [Context: "https://cla.opensource.microso..."]
| /github/workspace/README.md:28:110 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
| /github/workspace/README.md:29:50 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
| ------
| 2021-03-05 14:34:49 [INFO]   ---------------------------
| 2021-03-05 14:34:49 [INFO]   File:[/github/workspace/SECURITY.md]
| 2021-03-05 14:34:49 [ERROR]   Found errors in [markdownlint] linter!
| 2021-03-05 14:34:49 [ERROR]   Error code: 1. Command output:
| ------
| /github/workspace/SECURITY.md:3 MD041/first-line-heading/first-line-h1 First line in file should be a top level heading [Context: "## Security"]
| /github/workspace/SECURITY.md:17:243 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
| /github/workspace/SECURITY.md:21:1 MD007/ul-indent Unordered list indentation [Expected: 0; Actual: 2]
| /github/workspace/SECURITY.md:22:1 MD007/ul-indent Unordered list indentation [Expected: 0; Actual: 2]
| /github/workspace/SECURITY.md:23:1 MD007/ul-indent Unordered list indentation [Expected: 0; Actual: 2]
| /github/workspace/SECURITY.md:24:1 MD007/ul-indent Unordered list indentation [Expected: 0; Actual: 2]
| /github/workspace/SECURITY.md:25:1 MD007/ul-indent Unordered list indentation [Expected: 0; Actual: 2]
| /github/workspace/SECURITY.md:26:1 MD007/ul-indent Unordered list indentation [Expected: 0; Actual: 2]
| /github/workspace/SECURITY.md:27:1 MD007/ul-indent Unordered list indentation [Expected: 0; Actual: 2]
| ------
| 2021-03-05 14:34:49 [INFO]   ---------------------------
| 2021-03-05 14:34:49 [INFO]   File:[/github/workspace/SUPPORT.md]
| 2021-03-05 14:34:50 [ERROR]   Found errors in [markdownlint] linter!
| 2021-03-05 14:34:50 [ERROR]   Error code: 1. Command output:
| ------
| /github/workspace/SUPPORT.md:11 MD025/single-title/single-h1 Multiple top level headings in the same document [Context: "# Support"]
| /github/workspace/SUPPORT.md:15:95 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
| /github/workspace/SUPPORT.md:16:87 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
| /github/workspace/SUPPORT.md:19:100 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
| ------
| 2021-03-05 14:34:50 [INFO]   ----------------------------------------------
```

Comment (christoferlof) [reply .github/workflows/linter.yml:12]: The `VALIDATE_ALL_CODEBASE: false` on ln 25 configures the linter to only run on changes and not the complete codebase. 
Yes, whenever the Security and Support files are updated, these errors should to be fixed

**Comment by christoferlof** [comment]
Pushed update to security.md with intentional failure to trigger linter fail. 
image

**Comment by christoferlof** [comment]
Security.md updated to fix linting issues. Checks passed
image

**Review by marrobi — APPROVED** [review]

**Review by Lybecker — APPROVED** [review]

**Pull request merged**